packages/block/src/block.ts

```diff
@@ -459,9 +464,25 @@ export class Block {
       requestsRoot: reqRoot,
     }
 
+    const hasDepositRequests = depositRequests !== undefined && depositRequests !== null
+    const hasWithdrawalRequests = withdrawalRequests !== undefined && withdrawalRequests !== null
+    const requests =
+      hasDepositRequests || hasWithdrawalRequests ? ([] as CLRequest<CLRequestType>[]) : undefined
```

Comment:
Shouldn't this just check if there is a `requestsRoot`?

Reply:
well it can check that as well but I thought this check to be more thorough. though i have no strong opinions and can do requestroot check